Makefile

```diff
@@ -25,39 +86,120 @@ build: render
 	export $$(xargs < tests/cfg.env); \
 	python apps/api/src/hyp3_api/__main__.py
 
-install:
+.PHONY: install
+install: ## Installs required Python software with pip
 	python -m pip install -r requirements-all.txt
 
 files ?= job_spec/*.yml
 compute_env_file ?= job_spec/config/compute_environments.yml
 security_environment ?= ASF
 api_name ?= local
 cost_profile ?= DEFAULT
-render:
+.PHONY: render
+render: ## Render CloudFormation templates using HyP3 job specs and environment configs
 	@echo rendering $(files) for API $(api_name) and security environment $(security_environment); python apps/render_cf.py -j $(files) -e $(compute_env_file) -s $(security_environment) -n $(api_name) -c $(cost_profile)
 
-static: ruff openapi-validate cfn-lint
+.PHONY: static
+static: ## Perform static analysis with openapi-validate and ruff
+	ruff openapi-validate cfn-lint
 
-ruff:
+.PHONY: ruff
+ruff: ## Lint and format Python code with ruff
 	ruff check . && ruff format --diff .
 
-openapi-validate: render
+.PHONY: openapi-validate
+openapi-validate: render ## Validate openapi spec
 	openapi-spec-validator apps/api/src/hyp3_api/api-spec/openapi-spec.yml
 
-cfn-lint: render
+.PHONY: render cfn-lint
+cfn-lint: ## Lint CloudFormation templates
 	cfn-lint --info --ignore-checks W3002 E3008 --template `find . -name *-cf.yml`
 
-clean:
-	git ls-files -o -- apps | xargs rm; \
-	git ls-files -o -- lib/dynamo | xargs rm; \
-	git ls-files -o -- .pytest_cache | xargs rm; \
-	find ./ -empty -type d -delete; \
-	rm -f packaged.yml
+.PHONY: clean
+clean: ## Remove local build files
+	@{ \
+		if [ -f /.dockerenv ]; then \
+			echo \'make clean\' will remove your HyP3 deployment container; \
+		    echo Run \'make clean\' outside of you HyP3 deployment container; \
+		else \
+			git ls-files -o -- apps | xargs rm; \
+			git ls-files -o -- lib/dynamo | xargs rm; \
+			git ls-files -o -- .pytest_cache | xargs rm; \
+			find ./ -empty -type d -delete; \
+			rm -f packaged.yml; \
+			if command -v docker >/dev/null 2>&1; then \
+				if docker info >/dev/null 2>&1; then \
+					docker rmi -f ${DEPLOY_ENV_IMAGE_NAME}:latest 2>/dev/null || true; \
+				else \
+					echo "Docker is installed but the Docker engine is not running. Skipping deployment image cleanup."; \
+				fi; \
+			fi; \
+		fi; \
+	}
+
+.PHONY: image
+image: ## Create a containerized HyP3 deployment environment
+	docker build --pull -t ${DEPLOY_ENV_IMAGE_NAME}:latest -f Dockerfile .
+
+.PHONY: shell
+shell: ## Run a shell on Hyp3 deployment container (created with 'make image')
+	@{ \
+		$(SET_AWS_ACCOUNT_ENV_VARS) && \
+        printenv | grep AWS && \
+		$(DOCKER_RUN); \
+	}
+
+.PHONY: package
+package: ## Run 'make install && make build', and package CloudFormation artifacts
+	aws cloudformation package \
+		--template-file ./apps/main-cf.yml \
+		--s3-bucket "$$ARTIFACT_BUCKET_NAME" \
+		--output-template-file packaged.yml; \
+
+
+.PHONY: deploy
+deploy: ## Deploy HyP3 with AWS CloudFormation
```

Comment:
I think these might deprecate [.github/actions/deploy-hyp3](https://github.com/ASFHyP3/hyp3/tree/develop/.github/actions/deploy-hyp3). Though that may require writing a `.env` file as part of deployment...